ln-gateway/src/lib.rs

```diff
@@ -250,7 +250,8 @@ impl LnGateway {
     }
 
     async fn handle_balance_msg(&self) -> Result<Amount, LnGatewayError> {
-        self.federation_client.fetch_all_coins().await?;
+        let fetch_results = self.federation_client.fetch_all_coins().await;
+        fetch_results.into_iter().collect::<Result<Vec<_>, _>>()?;
```

Comment:
Even if there were multiple errors this will only return the first one (same as with the old `fetch_all_coins`) so maybe this should be changed to `-> Result<Amount, Vec<LnGatewayError>> { ... } ` @justinmoon ?